bazel_utils.go

```diff
@@ -0,0 +1,188 @@
+		var err error
+		f, err = os.CreateTemp("", "gcassert-object-code-inspection*.log")
+		if err != nil {
+			return err
+		}
+		mw = io.MultiWriter(pw, f)
+		defer fmt.Printf("See %s for full output.\n", f.Name())
+	}
+
+	disassembleCmdError := make(chan error, 1)
+	go func() {
+		defer func() {
+			_ = pw.Close()
+			if f != nil {
+				_ = f.Close()
+			}
+		}()
+		for _, path := range paths {
+			getPackageName := func(path string) string {
+				return path[strings.LastIndex(path, "/")+1:]
+			}
+
+			absPath, err := filepath.Abs(path)
+			if err != nil {
+				disassembleCmdError <- err
+				return
+			}
+			relPkgPath, err := filepath.Rel(workspacePath, absPath)
+			if err != nil {
+				disassembleCmdError <- err
+				return
+			}
+			objectFilePath := filepath.Join(bazelBin, relPkgPath, getPackageName(path)+".a")
+			args := []string{"tool", "objdump", objectFilePath}
+			objectFileDisassembleCmd := exec.Command("go", args...)
+			if debugMode {
+				objectFileDisassembleCmd.Stdout = mw
+				objectFileDisassembleCmd.Stderr = mw
+				// Log full 'objdump' command.
+				fmt.Fprintln(f, objectFileDisassembleCmd)
+			} else {
+				objectFileDisassembleCmd.Stdout = pw
+				objectFileDisassembleCmd.Stderr = pw
+			}
+
+			if err := objectFileDisassembleCmd.Run(); err != nil {
+				disassembleCmdError <- err
+				return
+			}
+		}
+		disassembleCmdError <- nil
+	}()
+
+	scanner := bufio.NewScanner(pr)
+
+	// callRegex is used to read a function identifier from a CALL line in the object file.
+	// Example call line from the object file:
+	/*
+	  mvcc.go:1135			0x66928a		94000000		CALL 0(PC)									[0:4]R_CALLARM64:github.com/cockroachdb/cockroach/pkg/storage.(*MVCCGetOptions).validate
+
+	*/
+	callRegex, err := regexp.Compile(`\s*(\S+)\s*\S*\s*\S*\s*CALL\s*\S*\s*(\S+)`)
+	if err != nil {
+		return err
+	}
+	// atleastMatchedOnce ensures that we are alerted if the regex is broken due to architecture / go version changes
+	// that change the structure of the disassembled object file.
+	var atleastMatchedOnce bool
+	for scanner.Scan() {
+		line := scanner.Text()
+		fnIdentifierMatch := callRegex.FindStringSubmatch(line)
+		if fnIdentifierMatch != nil {
+			atleastMatchedOnce = true
+			// See example call above for details on why this manipulation works.
+			currentFn := fnIdentifierMatch[2][strings.LastIndex(fnIdentifierMatch[2], ":")+1:]
+			if _, ok := bazelInlineSites[currentFn]; ok {
+				fileNameAndLineNum := fnIdentifierMatch[1]
+				message := "call was not inlined"
+				fmt.Fprintf(w, "%s: %s\t%s\n", message, fileNameAndLineNum, currentFn)
+			}
+		}
+	}
+
+	if err := <-disassembleCmdError; err != nil {
+		// If 'objdump' failed, return the error.
+		return err
+	}
+
+	if !atleastMatchedOnce {
+		errorMsg := `callRegex did not match any line in the disassembled object code. 
+This means it did not find any non-inlined function calls in the inspected object files. Ensure that the regex is still correct if this
```

Comment:
nit: we're not in crdb repo (where this is mandated in the style guide), but there are a few lines and comments (as well as in the other file) that are too long, consider wrapping them.